Add Sensei Home notice badge count

jom opened the pull request:
Fixes #5958 

### Changes proposed in this Pull Request

* Passes `max_age` down to all Notice data sources.
* Introduces `get_badge_count` into notices provider.
* Wires new method up to provide the count on the Home menu.

### Testing instructions
* Force notices by manipulating various versions of checked Sensei plugins.
* Check to make sure the notice count updates accordingly.

jom committed:
Make retry error a filterable value vs a parameter

jom committed:
Use max_age for all notice related data sources

jom committed:
Introduce batch_count method to provider

jom committed:
Wire up badge count

jom commented on includes/rest-api/class-sensei-rest-api-internal.php:
```diff
@@ -84,15 +84,16 @@ class Sensei_REST_API_Internal {
+		$sensei_home     = Sensei_Home::instance();
+		$remote_data_api = $sensei_home->get_remote_data_api();
 
+		$this->notices_provider     = $sensei_home->get_notices_provider();
```
@aaronfc I was wondering what your thoughts were on this. Potentially, we could have `Sensei_Home` manage instantiating all of this and then add a method to Sensei Home that was just `get_rest_api_controller()` that we could fit into this class. What do you think?

aaronfc replied:
Sounds good to me 👍

At some point it would be nice to look into adding some dependency injection solution... Until then having everything about Sensei Homr instantiated in the `Sensei_Home` sounds a good compromise :)

jom replied:
Cool be7c1e718c88082457bc00be0aefefc1f03cd703

aaronfc commented on includes/admin/class-sensei-home.php:
```diff
@@ -61,6 +69,15 @@ public function get_remote_data_api() {
+	/**
+	 * Gets the Sensei Home Notices provider instance.
+	 *
+	 * @return Sensei_Home_Remote_Data_API
```
Wrong return type!

jom replied:
Removed the method anyway

jom committed:
Move Sensei Home's providers to Sensei_Home

Makes `Sensei_Home` also responsible for producing its own REST API controller instance

jom committed:
Merge branch 'feature/new-onboarding-experience' into add/sensei-home-notice-count

jom committed:
Pass $max_age to local notice calls

renatho reviewed (approved):
Looks good and works well!
